pkg/ebpf/perf_ring_buffer.go

```diff
@@ -0,0 +1,76 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package ebpf
+
+import (
+	"sync"
+
+	"github.com/cilium/ebpf/ringbuf"
+
+	manager "github.com/DataDog/ebpf-manager"
+)
+
+var ringPool = sync.Pool{
+	New: func() interface{} {
+		return new(ringbuf.Record)
+	},
+}
+
+var _ EventHandler = new(RingBufferHandler)
+
+// RingBufferHandler wraps an eBPF ring buffer
+type RingBufferHandler struct {
+	RecordGetter func() *ringbuf.Record
+
+	dataChannel chan *DataEvent
+	lostChannel chan uint64
+	once        sync.Once
+	closed      bool
+}
+
+// NewRingBufferHandler creates a RingBufferHandler
+func NewRingBufferHandler(dataChannelSize int) *RingBufferHandler {
+	return &RingBufferHandler{
+		RecordGetter: func() *ringbuf.Record {
+			return ringPool.Get().(*ringbuf.Record)
+		},
+		dataChannel: make(chan *DataEvent, dataChannelSize),
+		// This channel is not really used in the context of ring buffers but
+		// it's here so `RingBufferHandler` and `PerfHandler` can be used
+		// interchangeably
+		lostChannel: make(chan uint64, 1),
+	}
+}
+
+// RecordHandler is the callback intended to be used when configuring PerfMapOptions
+func (c *RingBufferHandler) RecordHandler(record *ringbuf.Record, _ *manager.RingBuffer, _ *manager.Manager) {
+	if c.closed {
+		return
+	}
+
+	c.dataChannel <- &DataEvent{Data: record.RawSample, rr: record, CPU: -1}
```

Comment:
I'm setting CPU as `-1` for now, but I'd vote for removing it altogether from `DataEvent`.
NPM is the only other user of this code, and they will also start embedding CPU information in the event payload itself, which obviates the need for this field.